src/data/nat/prime_factorization.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2021 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell
+-/
+import data.nat.prime
+import data.nat.mul_ind
+
+/-!
+# Prime factorizations
+
+ `n.prime_factorization` is the finitely supported function `ℕ →₀ ℕ`
+ mapping each prime factor of `n` to its multiplicity in `n`.  For example, since 2000 = 2^4 * 5^3,
+  * `prime_factorization 2000 2` is 4
+  * `prime_factorization 2000 5` is 3
+  * `prime_factorization 2000 k` is 0 for all other `k : ℕ`.
+
+-/
+
+open nat finset list finsupp
+open_locale big_operators
+
+namespace nat
+
+/-- `n.prime_factorization` is the finitely supported function `ℕ →₀ ℕ`
+ mapping each prime factor of `n` to its multiplicity in `n`. -/
+noncomputable def prime_factorization (n : ℕ) : ℕ →₀ ℕ := (n.factors : multiset ℕ).to_finsupp
+
+lemma prime_factorization_count {n p : ℕ} : n.prime_factorization p = list.count p n.factors :=
+by simp [prime_factorization]
+
+/-- Every positive natural number has a unique prime factorization -/
+lemma prime_factorization_eq_iff {a b : ℕ} (ha : 0 < a) (hb : 0 < b) :
```

Comment:
This name could use `inj` somewhere. And you can state it using `set.inj_on` if you want to, which is quite practical because you get the `↔`, and `a ≠ b → a.prime_factorization ≠ b.prime_factorization` versions for free.
Also, you should ask for `≠ 0`, not `0 <`.

Reply:
Thanks, is the new lemma `factorization_inj` what you were looking for here?

Reply:
Which new lemma? 🤔

Reply:
Oh, sorry, I piled up a stack of commits and commented before pushing them!

Reply:
Yes it is!